src/service.js

```diff
@@ -591,13 +508,12 @@ function getServicePromiseOrNullInternal(holder, id) {
   if (s) {
     if (s.promise) {
       return s.promise;
-    } else if (s.obj) {
-      return s.promise = Promise.resolve(s.obj);
     } else {
-      dev().assert(s.build,
-          'Expected object, promise, or builder to be present');
-      s.obj = s.build();
-      return s.promise = Promise.resolve(s.obj);
+      if (!s.obj) {
+        // Instantiate service
```

Comment:
I think we can remove the conditional and change this comment to "Instantiate service if not already instantiated."